src/library/scala/collection/immutable/Vector.scala

```diff
@@ -466,24 +466,24 @@ override def companion: GenericCompanion[Vector] = Vector
       display5 = copyRange(display5, oldLeft, newLeft)
   }
 
-  private def zeroLeft(array: Array[AnyRef], index: Int): Unit = {
+  private[this] def zeroLeft(array: Array[AnyRef], index: Int): Unit = {
```

Comment:
In my experiment only avoiding the pre-null-ing of arrays had any performance impact (i.e. the array allocation and copying should be next to each other, otherwise the array is null-ed first).

Before:
```
VectorBenchmark.Slice.scala_persistent  thrpt   4193.769 ops/s
```

After:
```
VectorBenchmark.Slice.scala_persistent  thrpt  10306.275 ops/s
```

i.e. 2.5x faster, congrats @Ichoran